File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Source/Appendix A/VB/ServicesWithoutComponentsVB/App_Code/ProductType.vb
Imports Microsoft.VisualBasic
Imports System.Data
Imports System.Data.SqlClient
Public Class ProductType
    Public Function AddProductType(ByVal productID As Integer, ByVal productType As String) As Integer
        Dim productTypeID As Integer
        Dim connString As String = "Data Source=PUNEET-PC;Initial Catalog=componentapp;Integrated Security=True"
        Dim commandName As String = "AddProductTypes"
        Using conn As New SqlConnection(connString)
            conn.Open()
            Dim command As New SqlCommand(commandName, conn)
            command.CommandType = CommandType.StoredProcedure
            Dim paramProductID As New SqlParameter("@ProductID", SqlDbType.Int)
            paramProductID.Direction = ParameterDirection.Input
            paramProductID.Value = productID
            command.Parameters.Add(paramProductID)
            Dim paramProductType As New SqlParameter("@ProductType",
               SqlDbType.VarChar, 50)
            paramProductType.Direction = ParameterDirection.Input
            paramProductType.Value = productType
            command.Parameters.Add(paramProductType)
            Dim paramReturnValue As New SqlParameter("@@identity", SqlDbType.VarChar,
               250)
            paramReturnValue.Direction = ParameterDirection.ReturnValue
            command.Parameters.Add(paramReturnValue)
            command.ExecuteNonQuery()
            productTypeID = CInt(Fix(command.Parameters("@@identity").Value))
        End Using
        Return productTypeID
    End Function

End Class

File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Source/Appendix A/CS/ServicesWithoutComponentsCS/App_Code/ProductType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Data;
using System.Data.SqlClient;

/// <summary>
/// Summary description for ProductType
/// </summary>
public class ProductType
{
	public ProductType()
	{
		//
		// TODO: Add constructor logic here
		//
	}
    public int AddProductType(int productID, string productType)
    {
        int productTypeID;
        string connString = @"Data Source=PUNEET-PC;Initial Catalog=componentapp;Integrated Security=True";
        string commandName = "AddProductTypes";
        using (SqlConnection conn = new SqlConnection(connString))
        {
            conn.Open();
            SqlCommand command = new SqlCommand(commandName, conn);
            command.CommandType = CommandType.StoredProcedure;
            SqlParameter paramProductID = new SqlParameter("@ProductID",
             SqlDbType.Int);
            paramProductID.Direction = ParameterDirection.Input;
            paramProductID.Value = productID;
            command.Parameters.Add(paramProductID);
            SqlParameter paramProductType = new SqlParameter("@ProductType",
              SqlDbType.VarChar, 50);
            paramProductType.Direction = ParameterDirection.Input;
            paramProductType.Value = productType;
            command.Parameters.Add(paramProductType);
            SqlParameter paramReturnValue = new SqlParameter("@@identity",
              SqlDbType.VarChar, 250);
            paramReturnValue.Direction = ParameterDirection.ReturnValue;
            command.Parameters.Add(paramReturnValue);
            command.ExecuteNonQuery();
            productTypeID = (int)command.Parameters["@@identity"].Value;
        }
        return productTypeID;
    }

}